core/domain/prod_validators.py

```diff
@@ -2158,6 +2158,39 @@ def _get_external_id_relationships(cls, item):
                 'linked_skill_ids',
                 skill_models.SkillModel, item.linked_skill_ids)]
 
+    @classmethod
+    def _validate_inapplicable_skill_misconception_ids(cls, item):
+        """Validate that inapplicable skill misconception ids are valid.
+
+        Args:
+            item: ndb.Model. QuestionModel to validate.
+        """
+        inapplicable_skill_misconception_ids = (
+            item.inapplicable_skill_misconception_ids)
+        for skill_misconception_id in inapplicable_skill_misconception_ids:
+            skill_id, misconception_id = skill_misconception_id.split('-')
```

Comment:
Is it guaranteed that skill IDs cannot contain the character "-"?

If you're not sure, perhaps ":" is a better delimiter.

Reply:
We already use `tagged_skill_misconception_id` in the AnswerGroup which is of this format (with -). Also, looking at the id generation code fo skills:

https://github.com/oppia/oppia/blob/44fae73a01996313f80767377891f70740e66863/utils.py#L391-L397

It looks like skill ids are expected to be alphanumeric chars only. So it should not contain  "-".